### PR TITLE
Initial SSO updates for review

### PR DESCRIPTION
**Pull request by karensawrey**
Hi, @plaindocs! These are the main changes and additions I'd like to introduce to the SSO part. I hope to discuss some of it and further work during a sync.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -273,6 +273,7 @@
                   <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "Overview", 'integrations/sso' %></li>
                   <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "Okta", 'integrations/sso/okta' %></li>
                   <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "ADFS", 'integrations/sso/adfs' %></li>
+                  <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "GitHub", 'github_sso' %></li>
```
Comment (plaindocs) [inline app/views/layouts/application.html.erb:276]: :tada:

Comment (plaindocs) [inline app/views/layouts/application.html.erb:276]:
```suggestion
                  <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "GitHub", 'github-sso' %></li>
```

I should document this in the README somewhere, but Rails converts underscores in filenames to hypens in links

File: pages/integrations/sso.md.erb [hunk 2]
```diff
@@ -37,7 +37,20 @@ There are two ways to disable a provider:
...
+
+## Migrating from one SSO provider to another SSO provider    
+
+If you are the administrator of an organization within Buildkite with an existing SSO provider set up, and you want to switch to a different SSO provider, these are the steps you need to take:  
+
+1. [Add](/docs/integrations/sso#adding-sso) a new SSO provider, verify it, and allow login from both SSO providers. The users in your organization can continue to sign in and use the same user accounts within Buildkite as long as the emails stay the same.
+2. [Disable and remove](/docs/integrations/sso#disabling-and-removing-sso) the SSO provider you no longer need. If the user credentials (email) stay the same, this is all you need to migrate from one SSO provider to another.  
+
+<div class="Docs__note">
+  <p>If you are also changing the email provider, make sure that Buildkite users in your organization sign in to their existing accounts when performing single sign-on through the new provider. Otherwise, new user accounts will be created, and your organization will be billed for the same users twice.</p>
```
Comment (plaindocs) [inline pages/integrations/sso.md.erb:50]: Maybe rewording this to something friendlier like 

>  make sure that Buildkite users in your organization sign in to their existing accounts when performing single sign-on through the new provider to prevent your organization being billed twice for the same users

@jayco are we happy to state this in the docs?

File: pages/integrations/sso.md.erb [hunk 3]
```diff
@@ -50,6 +63,9 @@ Yes, we do.
 ### What happens if a person leaves our company?
 You will need to manually remove them from your Buildkite organization. This will not affect access to the user's personal account or any other organizations they are a member of.
 
+### Can I use different SSO providers for my Buildkite organization at the same time?
+Yes, as an admin you need to [add](/docs/integrations/sso#adding-sso) and verify a new SSO provider. Next, you need to allow login from both SSO providers in the [Organization settings](https://buildkite.com/organizations/-/sso). As long as the sign-in emails stay the same, the users in your organization can continue to sign in and use the same user accounts within Buildkite.
```
Comment (plaindocs) [inline pages/integrations/sso.md.erb:67]:
```suggestion
Yes, as an admin you need to [add and verify](/docs/integrations/sso#adding-sso) a new SSO provider. Next, you need to allow login from both SSO providers in the [Organization settings](https://buildkite.com/organizations/-/sso). As long as the sign-in emails stay the same, the users in your organization can continue to sign in and use the same user accounts within Buildkite.  
```

File: pages/integrations/sso/github_sso.md.erb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Single Sign-On with GitHub
+
+GitHub can be used as an SSO provider for your Buildkite organization. To complete this tutorial, you will need admin privileges for both Buildkite and your GitHub Organization.
```
Comment (plaindocs) [inline pages/integrations/sso/github_sso.md.erb:3]:
```suggestion
GitHub can be used as an SSO provider for your Buildkite organization. To complete this tutorial, you will need admin privileges for both the Buildkite organization and your GitHub organization.   
```

File: pages/integrations/sso/github_sso.md.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Single Sign-On with GitHub
+
+GitHub can be used as an SSO provider for your Buildkite organization. To complete this tutorial, you will need admin privileges for both Buildkite and your GitHub Organization.   
+
+### Step 1.  
+Setup the [Buildkite GitHub Application](https://github.com/apps/buildkite) for your GitHub organization. You need to install Buildkite for the GitHub Organization that you want to connect to Buildkite as an SSO provider.
```
Comment (plaindocs) [inline pages/integrations/sso/github_sso.md.erb:6]:
```suggestion
Set up the [Buildkite GitHub Application](https://github.com/apps/buildkite) for your GitHub organization. You need to install Buildkite for the GitHub organization that you want to connect to Buildkite as an SSO provider.  
```

set up as a verb
setup as a noun

<3

Comment (plaindocs) [reply pages/integrations/sso/github_sso.md.erb:6]: Maybe we can reduce this to 

```suggestion
Set up the [Buildkite GitHub Application](https://github.com/apps/buildkite) for the GitHub Organization that you want to connect to Buildkite as an SSO provider.  
```

what do you think?

File: pages/integrations/sso/github_sso.md.erb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+In your Buildkite Organization Settings' Single Sign On menu item, choose the GitHub provider:  
+<%= image "sso-settings.png", width: 872, height: 456, alt: "Screenshot of the Buildkite SSO Settings Page" %>
+
+You need to specify the name of your GitHub Organization for which you'd like to enable the SSO via GitHub and click Create Provider.  
+
+### Step 4. Perform a Test Login
+Follow the instructions to perform a test login. Performing a test login will verify that SSO is working correctly before you activate it for your organization members.  
+
+### Step 5. Enable the new SSO Provider
+Once you've performed a test login you can enable your provider. Activating SSO will not force a log out of existing users, but will cause all new or expired sessions to authorize through Okta before organization data can be accessed.
```
Comment (plaindocs) [inline pages/integrations/sso/github_sso.md.erb:21]: OKTA should be GitHub, right?

File: pages/integrations/sso/github_sso.md.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+If you need to edit or update your GitHub provider settings at any time, you will need to disable the provider first. For more information on disabling a provider, see the disabling SSO section of the SSO overview.
```
Comment (plaindocs) [inline pages/integrations/sso/github_sso.md.erb:23]: What do you think about deleting this whole sentence and just linking "disable the SSO provider"? 

> For more information on disabling a provider, see the disabling SSO section of the SSO overview.

**Review by plaindocs — COMMENTED** [review]
This is great! 
A few fixes and suggestions inline.

File: pages/integrations/sso.md.erb [hunk 8]
```diff
@@ -10,10 +10,10 @@ Buildkite supports the following SSO providers:
 
 * [Okta](/docs/integrations/sso/okta)
 * [ADFS](/docs/integrations/sso/adfs)
+* [GitHub](/docs/integrations/sso/github_sso)
```
Comment (plaindocs) [inline pages/integrations/sso.md.erb:13]:
```suggestion
* [GitHub](/docs/integrations/sso/github-sso)
```

File: pages/integrations/sso/github_sso.md.erb (hunk 6 above)
Comment (karensawrey) [reply pages/integrations/sso/github_sso.md.erb:21]: Ah, yes, used the piece from OKTA as a reference. The rest of the text if the same everywhere.

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -273,6 +273,7 @@
                   <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "Overview", 'integrations/sso' %></li>
                   <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "Okta", 'integrations/sso/okta' %></li>
                   <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "ADFS", 'integrations/sso/adfs' %></li>
+                  <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "GitHub", 'github-sso' %></li>
```
Comment (plaindocs) [inline app/views/layouts/application.html.erb:276]:
```suggestion
                  <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "GitHub", 'integrations/github-sso' %></li>
```

Comment (plaindocs) [reply app/views/layouts/application.html.erb:276]: Sorry, I'm not doing a great job of fixing these broken links.

**Comment by plaindocs** [comment]
This is super roundabout, but if you merge https://github.com/karensawrey/docs/pull/1 into your branch, it'll update this PR CI should pass.

File: pages/integrations/sso/adfs.md.erb [hunk 10]
```diff
@@ -6,15 +6,15 @@ Active Directory Federation Services (ADFS) can be used as an SSO provider for y
   <p>You can also set up SSO providers manually with GraphQL. See the <a href="/docs/integrations/sso/sso-setup-with-graphql">SSO Setup with GraphQL Guide</a> for detailed instructions and code samples.</p>
 </div>
 
-### Step 1. Create a Buildkite SSO Provider
+## Step 1. Create a Buildkite SSO Provider
 
 Click the Buildkite Organization Settings' Single Sign On menu item, then choose the ADFS provider from the available options:
 
-<%= image "sso-settings.png", width: 1458/2, height: 926/2, alt: "Screenshot of the Buildkite SSO Settings Page" %>
+<%= image "sso-settings.png", width: 872, height: 456, alt: "Screenshot of the Buildkite SSO Settings Page" %>
```
Comment (plaindocs) [inline pages/integrations/sso/adfs.md.erb:13]: Still trying to pin down exact screenshot stuff, but we probably need the larger screenshot / 2 here and elsewhere.

**Review by plaindocs — APPROVED** [review]
This looks great.

**Comment by plaindocs** [comment]
Merge it!
